src/Network/Cookie/Cookie.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+namespace Cake\Network\Cookie;
+
+use Cake\Core\InstanceConfigTrait;
+use Cake\I18n\Time;
+use Cake\Utility\Hash;
+use Cake\Utility\Security;
+use UnexpectedValueException;
+
+class Cookie
+{
+
+    const EXPIRES_FORMAT = 'U';
+    const TIMEOUT = '-1 month';
+
+    use InstanceConfigTrait;
+
+    /**
+     * Default config
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'path' => null,
+        'domain' => '',
+        'secure' => false,
+        'salt' => null,
+        'httpOnly' => false,
+        'encryption' => 'aes',
```

Comment:
@markstory even if encryption was a subclass request wouldn't know that while setting cookies. That information should be somehow pre-configured (like in CookieComponent) or kept inside cookie value, but that doesn't feel secure...